app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -100,18 +100,23 @@ class BrowserWebViewClient(
                 }
                 is SpecialUrlDetector.UrlType.Unknown -> {
                     Timber.w("Unable to process link type for ${urlType.url}")
-                    webView.loadUrl(webView.originalUrl)
+                    webView.loadUrl(webView.originalUrl, globalPrivacyControl.getHeaders())
                     false
                 }
                 is SpecialUrlDetector.UrlType.SearchQuery -> false
                 is SpecialUrlDetector.UrlType.Web -> {
                     if (requestRewriter.shouldRewriteRequest(url)) {
                         val newUri = requestRewriter.rewriteRequestWithCustomQueryParams(url)
-                        webView.loadUrl(newUri.toString())
+                        webView.loadUrl(newUri.toString(), globalPrivacyControl.getHeaders())
                         return true
                     }
                     if (isForMainFrame) {
-                        webViewClientListener?.willOverrideUrl(url.toString())
+                        if (globalPrivacyControl.isGpcActive()) {
```

Comment:
I've found some issues after introducing this change, so most probably this is affecting the callbacks we usually receive when loading a site.

What I tested?
- https://duckduckgo.github.io/privacy-test-pages/security/spoof-js-page-rewrite.html -> with GPC on this is not working now
- Login detection (fireproof) -> not working neither

I couldn't spot the problem. I will take a look later today.

Reply:
This is Solved now